pkg/apis/core/validation/validation.go

```diff
@@ -8172,3 +8192,12 @@ func validateLinuxContainerUser(linuxContainerUser *core.LinuxContainerUser, fld
 	}
 	return allErrors
 }
+
+func validateImageVolumeSource(imageVolume *core.ImageVolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(imageVolume.Reference) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("reference"), ""))
+	}
```

Comment:
this should be required for pods and allowed to be empty for deployments (other things containing podspec).

Please add tests for both cases.

Reply:
Fixed using `ResourceIsPod` option.